tokio/src/net/windows/named_pipe.rs

```diff
@@ -0,0 +1,1166 @@
+    ///
+    ///     let mut buf = [0u8; 4];
+    ///     let read = server.read_exact(&mut buf);
+    ///
+    ///     let ((), read) = tokio::try_join!(write, read)?;
+    ///
+    ///     println!("done reading and writing");
+    ///
+    ///     assert_eq!(read, 4);
+    ///     assert_eq!(&buf[..], b"ping");
+    /// }
+    /// # Ok(()) }
+    /// ```
+    pub fn access_outbound(&mut self, allowed: bool) -> &mut Self {
+        bool_flag!(self.open_mode, allowed, winbase::PIPE_ACCESS_OUTBOUND);
+        self
+    }
+
+    /// If you attempt to create multiple instances of a pipe with this flag,
+    /// creation of the first instance succeeds, but creation of the next
+    /// instance fails with [`ERROR_ACCESS_DENIED`].
+    ///
+    /// This corresponds to setting [`FILE_FLAG_FIRST_PIPE_INSTANCE`].
+    ///
+    /// [`ERROR_ACCESS_DENIED`]: crate::winapi::shared::winerror::ERROR_ACCESS_DENIED
+    /// [`FILE_FLAG_FIRST_PIPE_INSTANCE`]: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createnamedpipea#pipe_first_pipe_instance
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::io;
+    /// use tokio::net::windows::named_pipe::ServerOptions;
+    ///
+    /// const PIPE_NAME: &str = r"\\.\pipe\tokio-named-pipe-first-instance";
+    ///
+    /// # #[tokio::main] async fn main() -> io::Result<()> {
+    /// let mut builder = ServerOptions::new();
+    /// builder.first_pipe_instance(true);
+    ///
+    /// let server = builder.create(PIPE_NAME)?;
+    /// let e = builder.create(PIPE_NAME).unwrap_err();
+    /// assert_eq!(e.kind(), io::ErrorKind::PermissionDenied);
+    /// drop(server);
+    ///
+    /// // OK: since, we've closed the other instance.
+    /// let _server2 = builder.create(PIPE_NAME)?;
+    /// # Ok(()) }
+    /// ```
+    pub fn first_pipe_instance(&mut self, first: bool) -> &mut Self {
+        bool_flag!(
+            self.open_mode,
+            first,
+            winbase::FILE_FLAG_FIRST_PIPE_INSTANCE
+        );
+        self
+    }
+
+    /// Indicates whether this server can accept remote clients or not. This is
+    /// enabled by default.
+    ///
+    /// This corresponds to setting [`PIPE_REJECT_REMOTE_CLIENTS`].
+    ///
+    /// [`PIPE_REJECT_REMOTE_CLIENTS`]: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createnamedpipea#pipe_reject_remote_clients
+    pub fn reject_remote_clients(&mut self, reject: bool) -> &mut Self {
+        bool_flag!(self.pipe_mode, reject, winbase::PIPE_REJECT_REMOTE_CLIENTS);
+        self
+    }
+
+    /// The maximum number of instances that can be created for this pipe. The
+    /// first instance of the pipe can specify this value; the same number must
+    /// be specified for other instances of the pipe. Acceptable values are in
+    /// the range 1 through 254. The default value is unlimited.
+    ///
+    /// This value is only respected by the first instance that creates the pipe
+    /// and is ignored otherwise.
```

Comment:
First you say the same number must be specified, then you say it is ignored on subsequent calls. Which one is it?

Reply:
Oh gosh. This was added because testing (see the doctest) showed that the option is indeed ignored. [But MSDN claims otherwise](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createnamedpipea) which is where the other documentation was derived from. We ought to not contradict MSDN, because this might be a compatibility issue so I'll remove this.

Reply:
Actually nvm, the doctest does set it for all calls to create.